src/telemetry/metrics/mod.rs

```diff
@@ -200,7 +200,9 @@ impl Root {
 
         let process_metrics = process::Sensor::new()
             .map_err(|e| info!("{}", e))
-            .ok();
+            .ok()
+            .map(Mutex::new)
+            .map(Arc::new);
```

Comment:
I thought point-free syntax was [considered "too clever"](https://github.com/linkerd/linkerd2-proxy/pull/36#discussion_r207399016)? 😛

Reply:
in that case, we were using it on something that is effectively a method.  it's weird for me to see `.map(MyType::method)` as it seems to indicate that we're passing something to it, and while we are, technically, that's not really how i think about structures/calling things.  In this case, neither Mutex nor Arc take a `&self`, so I view it differently... the cleverness is al about the `&self` being passable to the thing

Reply:
Okay, that's reasonable!